### PR TITLE
Fix literal structure type

### PR DESCRIPTION
**Pull request by manggoguy**
#6 Fix codes

### CODE REVIEW

File: DebugIR.cpp [hunk 1]
```diff
@@ -409,8 +409,9 @@ class DIUpdater : public InstVisitor<DIUpdater> {
       return Builder.createUnspecifiedType("void");
     else if (T->isStructTy()) {
       // NOTE: where does DINodeArray come from?
+      StructType* ST = dyn_cast<StructType>(T);
```
Comment (vaivaswatha) [inline DebugIR.cpp]:
```suggestion
      StructType* ST = cast<StructType>(T);
```

File: DebugIR.cpp [hunk 2]
```diff
@@ -409,8 +409,9 @@ class DIUpdater : public InstVisitor<DIUpdater> {
...
       DICompositeType *S = Builder.createStructType(
-          LexicalBlockFileNode, T->getStructName(), FileNode,
+          LexicalBlockFileNode, !ST->isLiteral()? T->getStructName():"literal", FileNode,
```
Comment (vaivaswatha) [inline DebugIR.cpp]:
```suggestion
          LexicalBlockFileNode, ST->hasName()? T->getStructName():"literal", FileNode,
```

File: DebugIR.cpp [hunk 3]
```diff
@@ -431,8 +432,8 @@ class DIUpdater : public InstVisitor<DIUpdater> {
         DIType *ElDIType = getOrCreateType(ElType);
         DIType *MemType = Builder.createMemberType(
             LexicalBlockFileNode,
-            (T->getStructName().str() + "." +
-             std::to_string(tempNameCounter++)),
+            (!ST->isLiteral()?T->getStructName().str() + "." +
```
Comment (vaivaswatha) [inline DebugIR.cpp]:
```suggestion
            (ST->hasName()?T->getStructName().str() + "." +
```

Comment (vaivaswatha) [reply DebugIR.cpp]: Also could you run clang-format on the file once?

Comment (manggoguy) [reply DebugIR.cpp]: > Also could you run clang-format on the file once?

No, I didn't, but if it needed i'll do it.

Comment (vaivaswatha) [reply DebugIR.cpp]: > > Also could you run clang-format on the file once?
> 
> No, I didn't, but if it needed i'll do it.

Yes please.

Comment (manggoguy) [reply DebugIR.cpp]: >

> > > Also could you run clang-format on the file once?
> > 
> > 
> > No, I didn't, but if it needed i'll do it.
> 
> Yes please.

I ran clang-format, it shows message "no modified files to format"!

**Review by vaivaswatha — APPROVED** [review]

**Comment by vaivaswatha** [comment]
@manggoguy Thank you :pray:

**Pull request merged**